### PR TITLE
fix(delete): skip stale graph entities

### LINKED ISSUE #2049: Crash in UndoLog_DeleteNode when deleting stale startNode() result
## Summary

FalkorDB v4.18.3 can crash with SIGSEGV in `UndoLog_DeleteNode` when a query deletes a node, materializes that already-deleted node via `startNode(r)`, and then tries to delete the stale node value again.

This reproduces the same crash signature observed in FalkorDB/private#126:

```text
falkordb.so(UndoLog_DeleteNode+0x18)
falkordb.so(GraphHub_DeleteNodes+...)
RAX:0000000000000000
Exit code: 139
```

## Reproduction

```bash
docker run -d --name falkor-126 -p 6379:6379 falkordb/falkordb:v4.18.3
sleep 3
redis-cli -p 6379 GRAPH.QUERY g \
  "CREATE (a:EdgeCluster)-[r:R]->(a) WITH r, a DETACH DELETE a WITH startNode(r) AS ec DETACH DELETE ec"
```

## Actual behavior

The client receives:

```text
Error: Server closed the connection
```

The container exits with code 139 and logs show:

```text
Redis 8.6.2 crashed by signal: 11, si_code: 1
falkordb.so(UndoLog_DeleteNode+0x18)
falkordb.so(GraphHub_DeleteNodes+0x144)
RAX:0000000000000000
```

## Expected behavior

The query should not crash the server. Referencing a node that was already deleted in the same query pipeline should yield `NULL` or otherwise be rejected safely.

## Root cause

`AR_STARTNODE` / `AR_ENDNODE` call `Graph_GetNode()` but ignore its boolean return value. If the endpoint node was deleted, `Graph_GetNode()` returns false and leaves the returned `Node` with `attributes == NULL`. That invalid node can then be passed into `DETACH DELETE`, where `GraphHub_DeleteNodes()` calls `UndoLog_DeleteNode()` and the Rust FFI dereferences `n.attributes` unconditionally.

`Graph_EntityIsDeleted()` does not catch this because it returns false when `attributes == NULL`, treating that state as a reserved but not yet committed entity.

## Notes

A deterministic repro script was also captured locally under `repro/CRASH-F/` during investigation.

### PR DESCRIPTION
**Pull request by gkorland**
## Summary

Fixes a crash in `UndoLog_DeleteNode` when a stale node value with `attributes == NULL` is produced from a deleted relationship endpoint and later passed to `DELETE`.

## Changes

- Add `GraphEntity_CanModify` as the shared predicate for graph entities that are materialized and not already deleted.
- Use the shared predicate in `OpDelete` so stale/deleted entities are not collected as deletion candidates before `GraphHub_DeleteNodes`/`GraphHub_DeleteEdges`.
- Use the shared predicate in SET/REMOVE update paths so stale/deleted entities are treated as no-op updates.
- Add flow regression tests covering stale `startNode()`/`endNode()` DELETE and stale endpoint SET no-op behavior.

## Testing

- `./build.sh RUN_FLOW_TESTS=1 TEST=test_access_del_entity` ✅
- `git diff --check` ✅
- `make unit-tests` ⚠️ local run reached unit execution but failed existing `test_comparators` overflow assertions under the local GCC 15 / Redis 8.2.2 environment; unrelated to this change.

## Memory / Performance Impact

No expected memory impact. Runtime impact is limited to a shared materialized/not-deleted predicate while collecting DELETE/SET candidates.

## Related Issues

Fixes #2049
References FalkorDB/private#126

## Bug Fixes
* Fixed handling of endpoints produced by `startNode()` and `endNode()` functions when endpoints reference already-deleted nodes—`DETACH DELETE` and property update operations now behave correctly in these edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/FalkorDB/FalkorDB/pull/2050?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by gkorland** [comment]
@coderabbit review

File: src/graph/entities/graph_entity.c [hunk 1]
```diff
@@ -243,6 +243,13 @@ inline bool GraphEntity_IsDeleted
 	return Graph_EntityIsDeleted(e);
 }
 
+inline bool GraphEntity_CanModify
```
Comment (DvirDukhan) [inline src/graph/entities/graph_entity.c:246]: 1. I would rename it to `GraphEntity_IsValid`
2. `e->attributes != NULL` - I think it is hardening the condition. we might have entities without attribute set. please validate me

File: tests/flow/test_access_del_entity.py [hunk 2]
```diff
@@ -213,6 +213,57 @@ def test09_path_with_deleted_node(self):
         self.env.assertEquals(nodes[2].properties['v'], 'c')
         self.env.assertIn('C', nodes[2].labels)
 
+    def test10_delete_deleted_endpoint(self):
```
Comment (DvirDukhan) [inline tests/flow/test_access_del_entity.py:216]: I would add additional tests for nodes with attributes, to validate NULL or not NULL attribute set

**Comment by gkorland** [comment]
This PR is not relevant anymore as it's only relevant to the C code, which is no longer the active codebase (development has moved to the Rust rewrite). Closing accordingly.
